src/core/models/propulsion_model.py

```diff
@@ -0,0 +1,116 @@
+from src.core.models.model import ActuatorModel
+from src.core.parameters import Parameters
+from src.core.state.statetime import StateTime
+from typing import Dict, Any
+import numpy as np
+import cantera as ct
+import numpy as np
+#if is_propelling
+#    prop_model()
+#Prop_Model()
+#    augment the current state
+
+class PropulsionModel(ActuatorModel):
+    """Propagates location, time, based on propulsion model"""
+
+    def __init__(self, start_time: float, end_time: float, parameters: Parameters) -> None:
+      super().__init__(parameters)
+
+      self._t1 = start_time
+      self._t2 = end_time
+    
+    def evaluate(self, state_time: StateTime) -> Dict[str, Any]:
+        """Function that calculates propulsion force based on model
+
+      Args:
+          state_time (StateTime): The input statetime
+
+      Returns:
+          Dict[str, Any]: Thrust/Force (N)
+      """
+
+        ct.suppress_thermo_warnings(True)
+
+        d = 3 * 0.0254  # (m)
+        h = (3.968 + 1.618) * 0.0254  # (m)
+        d2 = 0.05 * 0.0254  # (m)
+        #d2mm = d2 * 1000
+        d3 = 0.305 * 0.0254  # (m)
+        #A1 = np.pi * d ** 2 / 4 #Cross-sectional area of the nozzle throat (m^2)
+        A2 = np.pi * d2 ** 2 / 4 #Cross-sectional area of the nozzle exit (m^2)
+        A3 = np.pi * d3 ** 2 / 4  # Cross-sectional area of the combustion chamber (m^2)
+        vc = h * np.pi * d ** 2 / 4
+        plim1 = 45 * 6894.76  # Pascals
+        plim2 = 100 * 6894.76  # Pascals
+        #RH2 = 4.124e3  # J/kg*K
+        #RO2 = 0.2598e3  # J/kg*K
+        RH2O = 0.4615e3  # J/kg*K
+
+        gas1 = ct.Solution('gri30.xml') #gri30 is cantera's database of gases with all the gas properties
+        gas2 = ct.Solution('gri30.xml')
+
+        gas1.TP = 298.0, plim1, 'H2:2,O2:1'
+        gas2.TP = 298.0, 1
+
+        gas1.equilibrate('SV')  # Equilibrate the initial gas mixture to fixed values of specific volume and entropy
+        combustor = ct.IdealGasReactor(gas1)
+        exhaust = ct.Reservoir(gas2) 
+        combustor.volume = vc 
+        network = ct.ReactorNet([combustor]) 
+        Choke = ct.MassFlowController(combustor, exhaust) # mass flow rate into combustion chamber
+        Taft=gas1.T
+        CpH2O = 2532.8  # Specific heat capacity of water vapor (J/kg*K)
+        CvH2O = 2071.2  # specific heat capacity of water vapor at constant volume (J/kg*K)
+        RH2O = CpH2O - CvH2O
+        gamma = CpH2O / CvH2O # Ratio of specific heats for water vapor
+        calc = (gamma + 1) / (2 * (gamma - 1))
+        arearatio = A2 / A3
+        int = arearatio * 1 / ((1 + (gamma - 1) / 2) ** calc)
```

Comment:
Please don't overwrite the standard python `int`

Reply:
Haha my bad I was trying to write a shorthand for intermediate and choose int. I'll fix that